server/midddleware/middleware.go

```diff
@@ -103,10 +105,10 @@ func Get(config *config.Config, tenantMgr *metadata.TenantManager, txMgr *transa
 		grpc_ratelimit.UnaryServerInterceptor(&RateLimiter{}),
 		grpc_auth.UnaryServerInterceptor(authFunction),
 		namespaceInitializer.NamespaceSetterUnaryServerInterceptor(),
-		grpctrace.UnaryServerInterceptor(grpctrace.WithServiceName(util.Service)),
-		grpc_logging.UnaryServerInterceptor(grpc_zerolog.InterceptorLogger(sampledTaggedLogger)),
-		validatorUnaryServerInterceptor(),
-		timeoutUnaryServerInterceptor(DefaultTimeout),
+	}
+
+	if config.DatadogTrace.Enabled {
+		unaryInterceptors = append(unaryInterceptors, ddTraceUnary())
 	}
 
 	if config.Metrics.Grpc.Enabled && config.Metrics.Grpc.ResponseTime {
```

Comment:
I wonder if we should put these Enabled flags under a single config structure and just case switch on them. WDYT?